bootstrap/util/configowner_test.go

```diff
@@ -20,113 +20,134 @@ import (
 	"context"
 	"testing"
 
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/pointer"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-func TestGetConfigOwnerSuccess(t *testing.T) {
+func TestGetConfigOwner(t *testing.T) {
+	g := NewGomegaWithT(t)
+
 	scheme := runtime.NewScheme()
-	if err := clusterv1.AddToScheme(scheme); err != nil {
-		t.Fatal("failed to register Cluster API objects to scheme")
-	}
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
 
-	myMachine := &clusterv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "my-machine",
-			Namespace: "my-ns",
-			Labels: map[string]string{
-				clusterv1.MachineControlPlaneLabelName: "",
+	t.Run("should get the owner when present (Machine)", func(t *testing.T) {
+		myMachine := &clusterv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-machine",
+				Namespace: "my-ns",
+				Labels: map[string]string{
+					clusterv1.MachineControlPlaneLabelName: "",
+				},
 			},
-		},
-		Spec: clusterv1.MachineSpec{
-			ClusterName: "my-cluster",
-			Bootstrap: clusterv1.Bootstrap{
-				DataSecretName: pointer.StringPtr("my-data-secret"),
+			Spec: clusterv1.MachineSpec{
+				ClusterName: "my-cluster",
+				Bootstrap: clusterv1.Bootstrap{
+					DataSecretName: pointer.StringPtr("my-data-secret"),
+				},
 			},
-		},
-		Status: clusterv1.MachineStatus{
-			InfrastructureReady: true,
-		},
-	}
+			Status: clusterv1.MachineStatus{
+				InfrastructureReady: true,
+			},
+		}
 
-	c := fake.NewFakeClientWithScheme(scheme, myMachine)
-	objm := metav1.ObjectMeta{
-		OwnerReferences: []metav1.OwnerReference{
-			{
-				Kind:       "Machine",
-				APIVersion: clusterv1.GroupVersion.String(),
-				Name:       "my-machine",
+		c := fake.NewFakeClientWithScheme(scheme, myMachine)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{
+					{
```

Comment:
Use `metav1.NewControllerRef`?

Reply:
This was copied from the earlier test, it's not an owner ref so I didn't change it